torch_brain/dataset/nested.py

```
@@ -0,0 +1,170 @@
+            transform: Optional transform that is applied to samples in :meth:`__getitem__`.
+
+        """
+        if isinstance(datasets, Mapping):
+            dataset_dict = datasets
+        elif isinstance(datasets, Iterable):
+            dataset_names = [ds.__class__.__name__ for ds in datasets]
+            if len(dataset_names) != len(set(dataset_names)):
+                raise ValueError(
+                    "Duplicate dataset class names found in provided datasets."
+                    " Please use a dictionary instead to specify dataset names explicitly."
+                )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)